### PR TITLE
feat(toc): v2 updates

### LINKED ISSUE #10512: [Component] ToC horizontal v2 WC
### Desktop

**V1**
Screenshot 2023-09-01 at 1 20 55 PM

---
**V2**
Screenshot 2023-09-01 at 1 23 00 PM

### Mobile

**V1**
Screenshot 2023-09-01 at 1 43 43 PM

---
**V2**
Screenshot 2023-09-01 at 1 44 20 PM

### Design changes
- [ ] Vertical active indicator
- [ ] Container color change
- [ ] Adds top and bottom borders to container
- [ ] Remove container drop shadow
- [ ] Keyboard controls (see specs / talk to Rich)
- [ ] mobile behavior change

### LINKED ISSUE #10514: [Component] ToC vertical v2 WC
[Figma spec](https://www.figma.com/file/rUQ0SVSdyC6uumEg6hQPS3/Table-of-contents-v2?type=design&node-id=15%3A15572&mode=design&t=aeP0EhkehW4Ljd9k-1)

### PR DESCRIPTION
**Pull request by ariellalgilmore**
### Related Ticket(s)

Closes #10512
Closes #10514

### Description

Updates ToC matching new v2 updates

### Changelog

**Changed**

- indicator now vertical next to text appearing in horizontal and mobile
- mobile is now just horizontal Toc

### CODE REVIEW

**Review by RichKummer — COMMENTED** [review]
@ariellalgilmore this is looking good! The new vertical indicator looks great and keyboard seems to be working well. A few comments as I was going through:

## Both horizontal and vertical
1. When I click on a selection in the `toc`, it takes me to the correct section but the title is covered by the `toc`. Can we add some additional rule to have the corresponding head visible when the `toc` has moved the user to the section?
Screenshot 2023-09-13 at 6 06 36 PM

2. When the `toc` is in the horizontal position (either set to horizontal or the window is too small for vertical) the `toc` should have a horizontal scroll – the mouse should ideally be able to scroll left and right like `tabs` currently. 

https://github.com/carbon-design-system/carbon-for-ibm-dotcom/assets/15643582/dcac40b8-5312-4be7-98f4-c19762ccb5d3

**Review by IgnacioBecerra — COMMENTED** [review]
Looking good! Just these comments:

File: packages/styles/scss/components/tableofcontents/_table-of-contents.scss [hunk 1]
```diff
@@ -350,72 +295,71 @@ $hover-transition-timing: 95ms;
...
+      position: absolute;
+      left: $spacing-08;
+      top: 0;
+      height: 100%;
+      width: $spacing-03;
+      z-index: 1;
+      background: linear-gradient(
+        to right,
+        $background,
+        rgba(255, 255, 255, 0)
```
[the remark this answers is not in the document]
Comment (ariellalgilmore) [reply packages/styles/scss/components/tableofcontents/_table-of-contents.scss:318]: the issue i was having was with the alpha value and with `linear-gradient` i couldn't get the tokens to work inside like `rgba($white, 0)`

**Review by IgnacioBecerra — APPROVED** [review]
LGTM, thanks @ariellalgilmore!

**Review by RichKummer — COMMENTED** [review]
@ariellalgilmore this is awesome! I noticed a minor issue which you can hopefully knock out, but if they need a separate PR let me know:

1. For the `horizontal` version, when clicking a selection the title is scrolled into view, which is great. But can we add a rule to shift the view so there's `16px` spacing at the top so the `toc` and the content are not flush together?

Screenshot 2023-09-21 at 2 22 20 PM

**Comment by ariellalgilmore** [comment]
@RichKummer just updated! deploy previews will be up soon 😄

**Review by RichKummer — APPROVED** [review]
LGTM! Thanks @ariellalgilmore 🎉

**Pull request merged**